### ISSUE TITLE
Feature Request: Autocomplete: Recognize scope

### ISSUE THREAD

**Issue by jim256** [hankinsoft/SQLPro#844]
**Is your feature request related to a problem? Please describe.**
If I have the following:
```
CREATE OR REPLACE TEMP TABLE _first_table AS

	SELECT
		*
	
	FROM
		PRODUCTS pro
;


CREATE OR REPLACE TEMP TABLE _second_table AS

	SELECT
		*
	
	FROM
		PROMOTIONS pro
	
	WHERE
		pro.
;
```

Autocomplete will suggest (sometimes) columns for the PRODUCTS table in the WHERE clause of _second_table, even though that table is not in the scope of the current statement, due to the fact that I used the same alias for the PROMOTIONS table.

**Describe the solution you'd like**
I'd like autocomplete to recognize the scope of where I'm typing, so it only autocompletes for things that would actually work when executed.

**Additional context**
SQLPro Studio 2022.71 (Build 110942.5)
macOS 12.5.1 (21G83)
Snowflake 6.31.1 (though I'd hope to see this implemented for at least Snowflake, Postgres, and MySQL)

**Comment by jaredx435k2d0** [comment]
Can't quite get it to do exactly that now that I try, either. If it happens again during real use, I'll try to capture it and post back.

That said, I do get autocomplete out of scope with the following (using sample Booktown connection):
<img width="611" alt="image" src="https://user-images.githubusercontent.com/111536894/194356282-d9bce5b8-176b-468f-ba3c-325045bcfad3.png">

```
CREATE TEMPORARY TABLE _first_table AS

	SELECT
		thisalias.
	
	FROM
		books AS thisalias
	
	LIMIT 10
;


CREATE TEMPORARY TABLE _second_table AS

	SELECT
		thisalias.
	
	FROM
		employee AS thisalias
	
	LIMIT 10
;


SELECT
	*

FROM
	doordash dd

WHERE
	thisalias.
```

[com.hankinsoft.osx.sqlprostudio 2022-10-06--15-33-42-039.log](https://github.com/hankinsoft/SQLPro/files/9726369/com.hankinsoft.osx.sqlprostudio.2022-10-06--15-33-42-039.log)

SQLPro Studio 2022.77 (Build 110949.5)
macOS 12.6 (21G115)

**Comment by hankinsoft** [comment]
Thanks for providing the code sample. I'm able to reproduce it and am looking into it.

**Comment by hankinsoft** [comment]
This build is now available (2022.81). If you get the chance, please give it a try and let me know if it has sorted this out.
